cmd/zoekt-sourcegraph-indexserver/main.go

```diff
@@ -241,6 +254,23 @@ func (s *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	repoTmpl.Execute(w, data)
 }
 
+func (s *Server) updateOptions(w http.ResponseWriter, r *http.Request) {
```

Comment:
Add a docstring to describe what this does. Also, include whether the behavior is "this will trigger an update to the index with the new options" or "this updates the options, which will get used the next time indexing runs".

Taking a step back, what will the overall contract be on how long it takes for the config update to be fully applied (such that end users will see the changes reflected in all search results)? Will it be all or nothing or gradual?